spring-cloud-stream/src/main/java/org/springframework/cloud/stream/config/BinderFactoryConfiguration.java

```diff
@@ -155,4 +178,21 @@ public BinderTypeRegistry binderTypeRegistry(ConfigurableApplicationContext conf
 		}
 		return parsedBinderConfigurations;
 	}
+
+	@ConditionalOnClass(name = "org.springframework.boot.actuate.health.HealthIndicator")
```

Comment:
This shouldn't move from an autoconfiguration to an imported configuration. I also think that in this case we need to make this a separate autoconfiguration (`@AutoConfigureBefore(EndpointAutoConfiguration.class)`) so that the added health indicators are properly injected.